File: android/app/src/main/kotlin/com/facepixel/app/Logger.kt
package com.facepixel.app

import android.util.Log

/**
 * Professional logging utility for FacePixel
 * Logs are only shown in debug builds
 */
object AppLogger {
    private const val PREFIX = "FacePixel"

    fun info(message: String, tag: String = "App") {
        if (BuildConfig.DEBUG) {
            Log.i("$PREFIX:$tag", message)
        }
    }

    fun debug(message: String, tag: String = "App") {
        if (BuildConfig.DEBUG) {
            Log.d("$PREFIX:$tag", message)
        }
    }

    fun warn(message: String, tag: String = "App") {
        if (BuildConfig.DEBUG) {
            Log.w("$PREFIX:$tag", message)
        }
    }

    fun error(message: String, tag: String = "App", throwable: Throwable? = null) {
        if (BuildConfig.DEBUG) {
            Log.e("$PREFIX:$tag", message, throwable)
        }
    }
}

File: web/logger.js
/**
 * Professional JavaScript logging utility
 * Logs are only shown in debug builds (when debugging is enabled)
 */

// Check if we're in debug mode
const DEBUG_MODE = typeof window !== 'undefined' &&
                   (window.location.hostname === 'localhost' ||
                    window.location.hostname === '127.0.0.1' ||
                    localStorage.getItem('DEBUG_MODE') === 'true');

const AppLogger = {
  /**
   * Log info message
   */
  info: function(message, tag = '') {
    if (DEBUG_MODE) {
      const tagStr = tag ? ` [${tag}]` : '';
      console.log(`[App]${tagStr}: ${message}`);
    }
  },

  /**
   * Log error message
   */
  error: function(message, tag = '', error = null) {
    if (DEBUG_MODE) {
      const tagStr = tag ? ` [${tag}]` : '';
      console.error(`[App]${tagStr} ERROR: ${message}`);
      if (error) {
        console.error('  Exception:', error);
        if (error.stack) {
          console.error('  Stack:', error.stack);
        }
      }
    }
  },

  /**
   * Log warning message
   */
  warning: function(message, tag = '') {
    if (DEBUG_MODE) {
      const tagStr = tag ? ` [${tag}]` : '';
      console.warn(`[App]${tagStr} WARNING: ${message}`);
    }
  },

  /**
   * Log debug message
   */
  debug: function(message, tag = '') {
    if (DEBUG_MODE) {
      const tagStr = tag ? ` [${tag}]` : '';
      console.log(`[App]${tagStr} DEBUG: ${message}`);
    }
  }
};
